### DISCUSSION TITLE
Why does Github not allow pushing directly to a public repo?

### DISCUSSION THREAD

**Discussion by lekiet1214** [community/community#40266] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

As far as I know, if I want to make contribution to a public repository, I will have to make a fork, push my changes to that fork then create a pull request to the original repository.
Why does it have to be that way? Why does Github not allow pushing directly to a public repository?

**Comment by airtower-luna** [comment, marked as answer]
It's basically a permission and consent system: If you could push directly, you could make changes regardless of what the repository owner wants. In your own fork you can do as you like (it's yours!), and if you make a PR the repository owner (or people they granted the permission to) can decide if they want to accept your changes.
